os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -478,10 +485,19 @@ void up_assert(const uint8_t *filename, int lineno)
 
 	is_kernel_fault = is_kernel_space((void *)assert_pc);
 
-#endif  /* CONFIG_BINMGR_RECOVERY */
+#endif  /* CONFIG_APP_BINARY_SEPARATION */
 
 	up_dumpstate();
 
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
```

Comment:
I don't think CONFIG_DEBUG_MM_HEAPINFO is mandatory for this.
We can't know the owner information which is almost mandatory for debug.
But without that config, it's helpful to know whether heap is corrupted or not.

Reply:
But, I have added heapinfo_check_corruption() API in heapinfo file. 
It will not get built when heapinfo is not enabled. If you think its ok to not have owner info at some times, then i will move this code to mallinfo file and call the api from other places.

Reply:
Location can't be a problem. We can move it to where we want.
We need to decide whether we will show memory corruption without debug information or not.
In my opinion, it will be definitely helpful. We can figure out the problem. And then we can do next step.
But if we don't show, then we should find what causes the assert.

Reply:
Done.